File: 2406-decode-the-message/2406-decode-the-message.cpp
class Solution {
public:
    string decodeMessage(string key, string message) {
        map<char, char> table;
        string decoded;

        for (auto& ch: key) {
            if (ch != ' ' && !table[ch]) {
                table[ch] = char('a' + table.size() - 1);
            }
        }

        for (auto& ch: message) {
            if (ch == ' ') {
                decoded += ch;
            } else {
                decoded += table[ch];
            }
        }

        return decoded;
    }
};

File: 2406-decode-the-message/2406-decode-the-message.py
class Solution:
    def decodeMessage(self, key: str, message: str) -> str:
        table = {}
        decoded = ""

        for k in key:
            if k != ' ' and k not in table:
                table[k] = chr(len(table) + 97)

        for m in message:
            decoded += m if m == ' ' else table[m]

        return decoded
        